### DISCUSSION TITLE
Images table UX

### DISCUSSION THREAD

**Discussion by rak-phillip** [rancher-sandbox/rancher-desktop#3017] [Research]
There's a few things I can point out regarding our table design that can be improved on:

1. Table header controls end up being rendering outside of the viewport. This can make it difficult to access the filters and any other controls we might add in the future
2. Columns should be resizable, and it should be easy to expose any truncated data in full 
3. Columns aren't fixed, making it easy to lose context when scrolled vertically

@manuelecarlini I think it makes sense to see if we can't improve on the design of our images table so that we can set a better standard for how tables should behave for features moving forward.

**Comment by rak-phillip** [comment]
Outcome of a conversation with Ken:

We could also add a column for age because people will be interested in images that they recently added

We will need to update sortable table so that we could take advantage of the new advanced filters

Can we consider hiding "Rancher Desktop" images by default?
* we will make this a preference that can be toggled

For long tags that might cause the column to grow, we can

1. truncate the tag with an ellipsis
2. show the entire tag in a popover on mouse hover or clicking the tag

**Comment by jandubois** [comment]
> Can we consider hiding "Rancher Desktop" images by default?

Yes, we should make it a toggle; not sure if it should be in Preferences, or just a switch on top (like the "Show muted" switch for diagnostics).

Just curious how we determine which image is a "Rancher Desktop" image, and which isn't.
